packages/next/build/webpack/plugins/wellknown-errors-plugin/parseRSC.ts

```diff
@@ -0,0 +1,90 @@
+import type { webpack } from 'next/dist/compiled/webpack/webpack'
+
+import { relative } from 'path'
+import { SimpleWebpackError } from './simpleWebpackError'
+
+export function formatRSCErrorMessage(
+  message: string
+): null | [string, string] {
+  if (message && /NEXT_RSC_ERR_/.test(message)) {
+    let formattedMessage = message
+    let formattedVerboseMessage = ''
+
+    // Comes from the "React Server Components" transform in SWC, always
+    // attach the module trace.
+    const NEXT_RSC_ERR_REACT_API = /.+NEXT_RSC_ERR_REACT_API: (.*?)\n/s
+    const NEXT_RSC_ERR_SERVER_IMPORT = /.+NEXT_RSC_ERR_SERVER_IMPORT: (.*?)\n/s
+    const NEXT_RSC_ERR_CLIENT_IMPORT = /.+NEXT_RSC_ERR_CLIENT_IMPORT: (.*?)\n/s
+
+    if (NEXT_RSC_ERR_REACT_API.test(message)) {
+      formattedMessage = message.replace(
+        NEXT_RSC_ERR_REACT_API,
+        `\n\nYou're importing a component that needs $1. It only works in a Client Component but none of its parents are marked with "client", so they're Server Components by default.\n\n`
+      )
+      formattedVerboseMessage =
+        '\n\nMaybe one of these should be marked as a "client" entry:\n'
+    } else if (NEXT_RSC_ERR_SERVER_IMPORT.test(message)) {
+      formattedMessage = message.replace(
+        NEXT_RSC_ERR_SERVER_IMPORT,
+        `\n\nYou're importing a component that imports $1. It only works in a Client Component but none of its parents are marked with "client", so they're Server Components by default.\n\n`
+      )
+      formattedVerboseMessage =
+        '\n\nMaybe one of these should be marked as a "client" entry:\n'
+    } else if (NEXT_RSC_ERR_CLIENT_IMPORT.test(message)) {
+      formattedMessage = message.replace(
+        NEXT_RSC_ERR_CLIENT_IMPORT,
+        `\n\nYou're importing a component that needs $1. That only works in a Server Component but one of its parents is marked with "client", so it's a Client Component.\n\n`
+      )
+      formattedVerboseMessage =
+        '\n\nOne of these is marked as a "client" entry:\n'
+    }
+
+    return [formattedMessage, formattedVerboseMessage]
+  }
+
+  return null
+}
+
+// Check if the error is specifically related to React Server Components.
+// If so, we'll format the error message to be more helpful.
+export function getRscError(
+  fileName: string,
+  err: Error,
+  module: any,
+  compilation: webpack.Compilation,
+  compiler: webpack.Compiler
+): SimpleWebpackError | false {
+  const formattedError = formatRSCErrorMessage(err.message)
+  if (!formattedError) return false
```

Comment:
Why not use null/undefined for billable error?

Reply:
Please check other formatters: https://github.com/vercel/next.js/blob/fbc4f3310f34ab52d69acc67dd1231a20c4b14d4/packages/next/build/webpack/plugins/wellknown-errors-plugin/